install.d/50-dracut.install

```diff
@@ -16,102 +16,64 @@ if [[ "$KERNEL_INSTALL_IMAGE_TYPE" = "uki" ]]; then
-    if [[ -z $KERNEL_INSTALL_UKI_GENERATOR || $KERNEL_INSTALL_UKI_GENERATOR == "dracut" ]]; then
-        # No uki generator preference set or we have been chosen
-        IMAGE="uki.efi"
-        UEFI_OPTS="--uefi"
-    elif [[ -z $KERNEL_INSTALL_INITRD_GENERATOR || $KERNEL_INSTALL_INITRD_GENERATOR == "dracut" ]]; then
-        # We aren't the uki generator, but we have been requested to make the initrd
-        IMAGE="initrd"
-        UEFI_OPTS="--no-uefi"
-    else
-        exit 0
-    fi
-elif [[ $KERNEL_INSTALL_LAYOUT == "bls" && -n $KERNEL_INSTALL_STAGING_AREA ]]; then
-    BOOT_DIR_ABS="$KERNEL_INSTALL_STAGING_AREA"
-    if [[ -z $KERNEL_INSTALL_INITRD_GENERATOR || $KERNEL_INSTALL_INITRD_GENERATOR == "dracut" ]]; then
-        IMAGE="initrd"
-        UEFI_OPTS="--no-uefi"
-    else
-        exit 0
-    fi
+if [[ "$KERNEL_INSTALL_INITRD_GENERATOR" = "dracut" ]]; then
+    # We are the initrd generator
+    IMAGE="initrd"
+    UEFI_OPTS="--no-uefi"
 else
-    # No layout information, use users --uefi/--no-uefi preference
-    UEFI_OPTS=""
-    if [[ -d $BOOT_DIR_ABS ]]; then
-        IMAGE="initrd"
-    else
-        BOOT_DIR_ABS="/boot"
-        IMAGE="initramfs-${KERNEL_VERSION}.img"
-    fi
+    exit 0
 fi
 
-ret=0
-
-case "$COMMAND" in
-    add)
-        if [[ $IMAGE == "uki.efi" ]]; then
-            IMAGE_PREGENERATED=${KERNEL_IMAGE%/*}/uki.efi
-        else
-            IMAGE_PREGENERATED=${KERNEL_IMAGE%/*}/initrd
-        fi
-        if [[ -f ${IMAGE_PREGENERATED} ]]; then
-            # we found an initrd or uki.efi at the same place as the kernel
-            # use this and don't generate a new one
-            [[ $KERNEL_INSTALL_VERBOSE == 1 ]] && echo \
-                "There is an ${IMAGE} image at the same place as the kernel, skipping generating a new one"
-            cp --reflink=auto "$IMAGE_PREGENERATED" "$BOOT_DIR_ABS/$IMAGE" \
-                && chown root:root "$BOOT_DIR_ABS/$IMAGE" \
-                && chmod 0600 "$BOOT_DIR_ABS/$IMAGE" \
-                && exit 0
-        fi
-
-        if [ -n "$KERNEL_INSTALL_CONF_ROOT" ]; then
-            if [ -f "$KERNEL_INSTALL_CONF_ROOT/cmdline" ]; then
-                read -r -d '' -a BOOT_OPTIONS < "$KERNEL_INSTALL_CONF_ROOT/cmdline"
-            fi
-        elif [[ -f /etc/kernel/cmdline ]]; then
-            read -r -d '' -a BOOT_OPTIONS < /etc/kernel/cmdline
-        elif [[ -f /usr/lib/kernel/cmdline ]]; then
-            read -r -d '' -a BOOT_OPTIONS < /usr/lib/kernel/cmdline
-        else
-            declare -a BOOT_OPTIONS
+if [[ "$KERNEL_INSTALL_UKI_GENERATOR" = "dracut" ]]; then
+    # We are chosen to generate the UKI as well as initrd
+    IMAGE="uki.efi"
+    UEFI_OPTS="--uefi"
+fi
 
-            read -r -d '' -a line < /proc/cmdline
-            for i in "${line[@]}"; do
-                [[ ${i#initrd=*} != "$i" ]] && continue
-                BOOT_OPTIONS+=("$i")
-            done
-        fi
+if [[ -f ${KERNEL_IMAGE%/*}/$IMAGE ]]; then
```

Comment:
AFAIU, this would make it noop when reinstalling `kernel-core`, which is not a good idea I think. I'd rather keep the previous behaviour.

Reply:
Yeah, you're right; it preserves the same behaviour. Sorry for the churn.